api/ProjectsAPI.py

```diff
@@ -0,0 +1,384 @@
+                    cursor.execute(uniq)
+                    results = cursor.fetchall()
+
+                    if len(results) > 0:
+                        cursor.close()
+                        disconnect(con)
+                        rsp_json = make_response(msg_uniq)
+                        rsp_json.headers.set('Content-Type', 'application/json')
+                        return (rsp_json, 403)
+                    else:
+                        query += "title = '" + content["title"] + "'"
+
+                elif i == "type":
+                    query += "type = '" + content["type"] + "'"
+                elif i == "description":
+                    query += "description = '" + content["description"] + "'"
+
+            query += " WHERE pid = " + pid + ";"
+            cursor.execute(query)
+            con.commit()
+
+            ## Retrieve the updated object
+            check = ("select pid, type, title, description "
+                    "from projects where pid = " + pid + ";")
+
+            cursor.execute(check)
+            row_headers=[x[0] for x in cursor.description]
+            row_headers.append('self')
+            results = cursor.fetchall()
+
+            project_list = []
+
+            if len(results) > 0:
+                for row in results:
+                    url = (request.url,)
+                    new_tup = row + url
+                    project_list.append(dict(zip(row_headers, new_tup)))
+
+            ## Return the new object JSON
+            rsp_json = make_response(json.dumps(
+                project_list, 
+                indent=2,
+                separators=(',', ':'),
+                default=str))
+            rsp_json.headers.set('Content-Type', 'application/json')
+
+            cursor.close()
+            disconnect(con)
+            return (rsp_json, 201)
+
+        except:
+            if cursor:
+                cursor.close()
+            if con:
+                disconnect(con)
+            rsp_json = make_response(msg_fail)
+            rsp_json.headers.set('Content-Type', 'application/json')
+            return (rsp_json, 500)
+ 
+## -------------------------------------------##
+## Delete a Project
+## -------------------------------------------##
+@projects_api.route('/projects/<pid>', methods=['DELETE'])
+def deleteProject(pid):
+    try:
+        # Open DB connection and see if pid exissts
+        con = dbconnect()
+        cursor = con.cursor()
+        query = ("SELECT pid from projects WHERE pid = '" + pid + "';")
+        cursor.execute(query)
+        results = cursor.fetchall()
+
+        # If no results, then no pid exists
+        if len(results) <= 0:
+            con.commit()
+            cursor.close()
+            rsp_json = make_response(msg_none)
+            rsp_json.headers.set('Content-Type', 'application/json')
+            return (rsp_json, 404)
+        else:
+            # Delete Project
+            query = ("DELETE from projects WHERE pid = " + pid + ";")
```

Comment:
Won't this fail a foreign key constraint with images? Do we need to add something like `ON DELETE CASCADE` to projects?